x/httpproxy/connect_handler.go

```diff
@@ -76,7 +76,8 @@ func (h *connectHandler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http
 
 	// Dial the target.
 	transportConfig := proxyReq.Header.Get("Transport")
-	dialer, err := config.WrapStreamDialer(h.dialer, transportConfig)
+
+	dialer, err := config.WrapStreamDialer(&config.StreamDialer{StreamDialer: h.dialer, Config: ""}, transportConfig)
```

Comment:
Why are you changing this?

Reply:
with the changes to `config.go`, `WrapStreamDialer` is now taking `StreamDialer` type as input while `h.dialer` is `transport.StreamDialer` type. That's why I need to do this casting otherwise the code breaks.

Reply:
Typically `WrapStreamDialer` is not directly called outside of `config.go` package and the most common use case is to use `config.NewStreamDialer`.